index.prod.mjs

```diff
@@ -0,0 +1,230 @@
+/* eslint-disable no-debugger */
+/* eslint-disable node/no-deprecated-api */
+/* eslint-disable camelcase */
+/* eslint-disable promise/param-names */
+import http from 'http';
+import https from 'https';
+import url from 'url';
+import path from 'path';
+import fs from 'fs';
+import express from 'express';
+import wsrtc from 'wsrtc/wsrtc-server.mjs';
+import metaversefile from 'metaversefile/plugins/rollup.js';
+import glob from 'glob';
+
+Error.stackTraceLimit = 300;
+const cwd = process.cwd();
+
+const isProduction = process.argv[2] === '-p';
+process.env.MODULE_URL = process.env.MODULE_URL || 'https://local.webaverse.com/assets/';
+const totum = metaversefile();
+
+const _isMediaType = p => /\.(?:png|jpe?g|gif|svg|glb|mp3|wav|webm|mp4|mov)$/.test(p);
+
+const _tryReadFile = p => {
+  try {
+    return fs.readFileSync(p);
+  } catch (err) {
+    // console.warn(err);
+    return null;
+  }
+};
+const certs = {
+  key: _tryReadFile('./certs/privkey.pem') || _tryReadFile('./certs-local/privkey.pem'),
+  cert: _tryReadFile('./certs/fullchain.pem') || _tryReadFile('./certs-local/fullchain.pem'),
+};
+
+function makeId(length) {
+  let result = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length));
+  }
+  return result;
+}
+
+function dynamicImporter(o, req, res, next) {
+  try {
+    const loadUrl =   o.pathname.slice(o.pathname.lastIndexOf('/@import'),o.pathname.length).replace('/@import', '');
+    const fullUrl = req.protocol + '://' + req.get('host') + loadUrl;
+    const reqURL = new URL(fullUrl);
+
+    /** Check intiator */
+    if (req.headers['sec-fetch-dest'] === 'script') {
+      totum.resolveId(loadUrl, reqURL.href).then(id => {
+        /** ID might have /@proxy/ in the start that needs to be trimmed */
+        if (!id) {
+          res.status(500);
+          return res.end('Failed to load');
+        }
+
+        id = id.replace('/@proxy/', '');
+
+        totum.load(id).then(({code, map}) => {
```

Comment:
Why isn't this async? There is a try catch block but then there is this async part with a different flow.

Reply:
Resolved promise/async usage.